text/0092-replay-issue-creation.md

```diff
@@ -17,43 +17,66 @@ These issues can only be detected on the SDK. The Replay back-end will never hav
 
 # Options Considered
 
-### Option 1: SDK Generates Issues Through a Generic Interface
+### Option 1: SDK Creates Issues Through the Error Interface
 
-When the SDK encounters a "replay issue" it will make an HTTP request to a generic interface which will handle the issue creation process.
+When the SDK encounters a "slow click" it will use `sentry.captureException(slowClick)` to create an error issue on the Issues platform.
 
 **Pros:**
 
-1. We can be tested in isolation without impacting existing production services.
-2. We can expose these new features in a more product agnostic way.
-   - This would allow us to reach much larger customer segments when compared to Replay customers.
+- Significantly larger customer reach (i.e. all javascript SDK customers).
+- Opportunity to upsell customers on Session Replay from a "slow click" event.
+- No code changes required on the Session Replay back end.
+- No code changes required by the Issues team.
 
 **Cons:**
 
-1. Uses errors quota.
-2. Unclear if we're able to use dynamic thresholds.
-   - E.g. "Experienced 10 occurences in the past hour".
-3. The replay containing the issue could not be sampled.
+- Requires SDK changes.
+- Can not change issue sampling parameters without an SDK update.
+  - Or can you?
+  - What if we sample within the error processing pipeline on the back-end?
+  - If error processing is functionally immutable (i.e. there is no organizational will-power to allow product teams to sample within the error pipeline) then we will not be able to sample on the back-end.
 
-### Option 2: SDK Generates Issues Through a Session-Replay Specific Interface
+### Option 2: SDK Creates Issues Through an Issues HTTP Endpoint
 
-The SDK publishes a "replay issue" to the Replay back-end. The back-end will decide to process the issue or not.
+When the SDK encounters a "slow click" it will use `http.post("/issues", data=slowClick)` to create an error issue on the Issues platform.
```

Comment:
Not sure this is feasible. Given we'd need the whole mechanism of retry/backpressure, serialization, etc. It would need to be a new `envelopeType` and use the same `captureEnvelope` mechanism of the SDK instead.

Reply:
I'm not sure the exact implementation details.  This was an idea proposed by the Issues team.  A generic /issues endpoint which product teams could make requests to.  My http client psuedo-code shouldn't be interpreted as production code.  I'm only trying to indicate that there's a new /issues HTTP endpoint which is submitted to by the SDK.

Reply:
Would this be rolled out on Relay? How do we handle all the ingestion scaling, PoP's etc? The endpoint could be used internally by the consumer if that helps somehow, but to be called by the clients, I imagine we'll need to stick to `/envelope` and Relay. But happy keep this option here, just raising my concerns.

Reply:
Not sure of the implementation details.  I would imagine the endpoint would be added to Relay and messages forwarded to the Issues Kafka consumer.